src/transaction/cleanup_state.cpp

```diff
@@ -25,10 +25,13 @@ void CleanupState::CleanupEntry(UndoFlags type, data_ptr_t data) {
 		// destroy the backed up entry: it is no longer required
 		assert(catalog_entry->parent);
 		if (catalog_entry->parent->type != CatalogType::UPDATED_ENTRY) {
-			if (!catalog_entry->parent->child->deleted) {
+			if (catalog_entry->parent->deleted) {
```

Comment:
I saw the change in this line yesterday, and I still don't understand it: why do we go from checking that the child of the parent is not deleted to checking that the parent is deleted, but still erase the child object from the dependency manager?

Reply:
That's a good point, I don't really remember why I changed it to this. Looking at it again the correct code is `if (!catalog_entry->deleted)` (which is equivalent to the original code, since `parent->child` is a nop operation).